shared/fs/popups/row-action-popup-container.js

```diff
@@ -159,14 +167,24 @@ const mergeProps = (stateProps, dispatchProps) => {
     lastWriter: pathItem.lastWriter.username,
     name: pathItem.name,
     size: pathItem.size,
+    needLoadMimeType: pathItem.type === 'file' && pathItem.mimeType === '',
     childrenFolders,
     childrenFiles,
     itemStyles,
     menuItems,
     onHidden,
+    loadMimeType: () => _loadMimeType(path),
   }
 }
 
-export default compose(connect(mapStateToProps, mapDispatchToProps, mergeProps), setDisplayName('Popup'))(
-  Popup
-)
+export default compose(
+  connect(mapStateToProps, mapDispatchToProps, mergeProps),
+  lifecycle({
+    componentDidMount() {
```

Comment:
Is there a risk of a race here? I just want to make sure that we've loaded the mime type before rendering the popup.

Reply:
Yes, but it's no harm. In fact, the first time you see this popup on an image, it won't show a "Save" button initially, and the Save button pops up instantly. We could probably move this up to before `routeAppend` if this turns out to be annoying.